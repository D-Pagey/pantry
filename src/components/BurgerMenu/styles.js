import styled from 'styled-components';
import { NavLink as Link } from 'react-router-dom';

export const NavLink = styled(Link)`
    color: white;
    text-decoration: none;

    &:hover {
        color: orange;
    }
`;

export const MenuStyles = {
    bmBurgerButton: {
        position: 'fixed',
        width: '36px',
        height: '30px',
        right: '36px',
        top: '36px'
    },
    bmBurgerBars: {
        background: '#373a47'
    },
    bmBurgerBarsHover: {
        background: '#a90000'
    },
    bmCrossButton: {
        height: '24px',
        width: '24px'
    },
    bmCross: {
        background: '#bdc3c7'
    },
    bmMenuWrap: {
        position: 'fixed',
        height: '100%'
    },
    bmMenu: {
        background: '#373a47',
        padding: '2.5em 1.5em 0',
        fontSize: '1.15em'
    },
    bmMorphShape: {
        fill: '#373a47'
    },
    bmItemList: {
        color: '#b8b7ad',
        padding: '0.8em'
    },
    bmItem: {
        margin: '0 0 1rem'
    },
    bmOverlay: {
        background: 'rgba(0, 0, 0, 0.3)'
    }
};
