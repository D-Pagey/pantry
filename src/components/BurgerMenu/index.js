import React, { useContext, useState } from 'react';
import { slide as Menu } from 'react-burger-menu';
import { FirebaseContext } from '../ProviderFirebase';
import * as S from './styles';

const BurgerMenu = () => {
    const [isOpen, setIsOpen] = useState(false);
    const { isAuthed, signOut } = useContext(FirebaseContext);

    const handleStateChange = (state) => setIsOpen(state.isOpen);
    const closeMenu = () => setIsOpen(false);
    const menuSignOut = () => {
        signOut();
        closeMenu();
    };

    return (
        <Menu
            isOpen={isOpen}
            onStateChange={handleStateChange}
            styles={S.MenuStyles}
            disableAutoFocus
            right
        >
            <S.NavLink to="/" onClick={closeMenu}>
                Home
            </S.NavLink>
            {isAuthed ? (
                <>
                    <S.NavLink to="/add" onClick={closeMenu}>
                        Add Food
                    </S.NavLink>
                    <S.NavLink
                        to="/profile"
                        onClick={closeMenu}
                        data-testid="burgerMenuProfileLink"
                    >
                        Profile
                    </S.NavLink>
                    <S.Button onClick={menuSignOut} data-testid="burgerMenuSignOut">
                        Sign Out
                    </S.Button>
                </>
            ) : (
                <S.NavLink to="/sign-in" onClick={closeMenu}>
                    Sign In
                </S.NavLink>
            )}
        </Menu>
    );
};

export default BurgerMenu;
