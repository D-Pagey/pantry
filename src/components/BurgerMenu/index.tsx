import React, { FC, useContext, useState } from 'react';
import { slide as Menu } from 'react-burger-menu';

import { AuthContext } from '../ProviderAuth';
import * as S from './styles';

export const BurgerMenu: FC = () => {
    const [isOpen, setIsOpen] = useState(false);
    const { user, signOut } = useContext(AuthContext);

    const handleStateChange = (state: { isOpen: boolean }): void => setIsOpen(state.isOpen);
    const closeMenu = (): void => setIsOpen(false);
    const menuSignOut = (): void => {
        signOut();
        closeMenu();
    };

    return (
        <Menu isOpen={isOpen} onStateChange={handleStateChange} styles={S.MenuStyles} disableAutoFocus right>
            <S.NavLink to="/" onClick={closeMenu}>
                Home
            </S.NavLink>
            {user ? (
                <>
                    <S.NavLink to="/add" onClick={closeMenu}>
                        Add Food
                    </S.NavLink>
                    <S.NavLink to="/settings" onClick={closeMenu} data-testid="burgerMenuSettingsLink">
                        Settings
                    </S.NavLink>
                    <S.Button onClick={menuSignOut} data-testid="burgerMenuSignOut">
                        Sign Out
                    </S.Button>
                </>
            ) : (
                <S.NavLink to="/sign-in" onClick={closeMenu}>
                    Sign In
                </S.NavLink>
            )}
        </Menu>
    );
};
