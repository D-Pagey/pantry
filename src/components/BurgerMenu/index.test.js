import React from 'react';
import BurgerMenu from '.';

describe('BurgerMenu component', () => {
    it('should render', () => {
        const { container } = render(<BurgerMenu />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it.todo('should close on click');
});
