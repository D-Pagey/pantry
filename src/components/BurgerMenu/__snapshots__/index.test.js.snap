// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu component should render 1`] = `
.c0 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0 0 1rem;
}

.c0:hover {
  color: orange;
}

<div>
  <div
    class="bm-overlay"
    style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
  />
  <div
    class="bm-menu-wrap"
    id=""
    style="position: fixed; right: 0px; z-index: 1100; width: 300px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
  >
    <div
      class="bm-menu"
      style="height: 100%; box-sizing: border-box; overflow: auto; background: rgb(55, 58, 71); padding: 2.5em 1.5em 0px; font-size: 1.15em;"
    >
      <nav
        class="bm-item-list"
        style="height: 100%; color: rgb(184, 183, 173); padding: 0.8em; display: flex; flex-direction: column;"
      >
        <a
          aria-current="page"
          class="c0 bm-item active"
          href="/"
          style="display: block;"
          tabindex="-1"
        >
          Home
        </a>
        <a
          class="c0 bm-item"
          href="/sign-in"
          style="display: block;"
          tabindex="-1"
        >
          Sign In
        </a>
      </nav>
    </div>
    <div>
      <div
        class="bm-cross-button"
        style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
      >
        <span
          style="position: absolute; top: 6px; right: 14px;"
        >
          <span
            class="bm-cross"
            style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg); background: rgb(189, 195, 199);"
          />
          <span
            class="bm-cross"
            style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg); background: rgb(189, 195, 199);"
          />
        </span>
        <button
          style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
          tabindex="-1"
        >
          Close Menu
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      class="bm-burger-button"
      style="z-index: 1000; position: fixed; width: 36px; height: 30px; right: 36px; top: 36px;"
    >
      <span>
        <span
          class="bm-burger-bars"
          style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1; background: rgb(55, 58, 71);"
        />
        <span
          class="bm-burger-bars"
          style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1; background: rgb(55, 58, 71);"
        />
        <span
          class="bm-burger-bars"
          style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1; background: rgb(55, 58, 71);"
        />
      </span>
      <button
        style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
      >
        Open Menu
      </button>
    </div>
  </div>
</div>
`;
